### ISSUE TITLE
GPU support and batch size

### ISSUE THREAD

**Issue by XzwHan** [nnuncert/nnuncert#2]
Hi,

I'm currently applying "PBP" and "PNN-E" from your repo for running UCI regression tasks, however I wasn't able to increase the training speed after I increased batch_size when calling model.fit(), and after configuring tensorflow-gpu and made it available in the environment, I wasn't able to speed up the training either. I was wondering if the current code base supports GPU, and if there's a different argument that I shall set the training batch size.

Many thanks!

**Comment by pjoachims** [comment]
Hej, 

most methods here are implemented using tensorflow and all parameters passed to `fit` method will be passed onto the underlying tensorflow models (probably you used this already). Further, this should ensure that all those methods should run on a GPU, if present. However, for small models the **overhead of using a GPU** (see [here](https://stackoverflow.com/questions/55749899/training-a-simple-model-in-tensorflow-gpu-slower-than-cpu)) may lead to higher runtimes compared to CPU. I did make the same oberservation here.

Chaning the batch-size does seem to work properly for the PNN-E:
| Deep Ensemble (N=5, epochs=100) | CPU (Google Colab) |
|---------------------------------|---------|
| batch_size=4                    | 99.98 s |
| batch_size=32                   | 18.71 s |
| batch_size=100                  | 13.08 s |

For **PBP**, I basically used the theano code provided from [here](https://github.com/HIPS/Probabilistic-Backpropagation). As they did not implement a `batch_size` parameter, setting the batch size does not affect anything here. Under the hood, it looks like:
```python
class PBPModel():
    ...
    def fit(x_train, y_train, epochs, *args, **kwargs):
	...
        self.pbp = PBP_net(x_train, y_train, self.n_hidden_units, n_epochs=epochs, normalize=False)
```

Hope this helps and sorry for the late reply.

**Comment by XzwHan** [comment]
Thank you very much for your reply! I was able to modify the corresponding tensorflow code to run the experiments on UCI datasets. My work is included in the NeurIPS 2022 [proceedings](https://proceedings.neurips.cc/paper_files/paper/2022/file/72dad95a24fae750f8ab1cb3dab5e58d-Paper-Conference.pdf), which recently became public. I have cited your repository along with the URL. I would like to express my sincere gratitude to you for creating this repository, which has enabled me to re-run the regression tasks using various BNN models.
